site/docs/latest/functions/overview.md

```diff
@@ -52,51 +55,200 @@ $ bin/pulsar-functions localrun \
+Local run mode | The function runs in your local environment, for example on your laptop
+Cluster mode | The function runs *inside of* your Pulsar cluster, on the same machines as your Pulsar {% popover brokers %}
+
+### Local run mode {#local-run}
+
+If you run a Pulsar Function in **local run** mode, it will run on the machine from which the command is run (this could be your laptop, an [AWS EC2](https://aws.amazon.com/ec2/) instance, etc.). Here's an example [`localrun`](../../CliTools#pulsar-admin-functions-localrun) command:
+
+```bash
+$ bin/pulsar-admin functions localrun \
+  --py myfunc.py \
+  --className myfunc.SomeFunction \
+  --inputs persistent://sample/standalone/ns1/input-1 \
+  --output persistent://sample/standalone/ns1/output-1
+```
+
+By default, the function will connect to a Pulsar cluster running on the same machine, via a local {% popover broker %} service URL of `pulsar://localhost:6650`. If you'd like to use local run mode to run a function but connect it to a non-local Pulsar cluster, you can specify a different broker URL using the `--brokerServiceUrl` flag. Here's an example:
+
+```bash
+$ bin/pulsar-admin functions localrun \
+  --brokerServiceUrl pulsar://my-cluster-host:6650 \
+  # Other function parameters
+```
+
+### Cluster run mode {#cluster-run}
+
+When you run a Pulsar Function in **cluster mode**, the function code will be uploaded to a Pulsar {% popover broker %} and run *alongside the broker* rather than in your [local environment](#local-run). You can run a function in cluster mode using the [`create`](../../CliTools#pulsar-admin-functions-create) command. Here's an example:
+
+```bash
+$ bin/pulsar-admin functions create \
+  --py myfunc.py \
+  --className myfunc.SomeFunction \
+  --inputs persistent://sample/standalone/ns1/input-1 \
+  --output persistent://sample/standalone/ns1/output-1
+```
+
+This command will upload `myfunc.py` to Pulsar, which will use the code to start one [or more](#parallelism) instances of the function.
+
+### Parallelism
+
+By default, only one **instance** of a Pulsar Function runs when you create and run it in [cluster run mode](#cluster-run). You can also, however, run multiple instances in parallel. You can specify the number of instances when you create the function, or update an existing single-instance function with a new parallelism factor.
+
+This command, for example, would create and run a function with a parallelism of 5 (i.e. 5 instances):
+
+```bash
+$ bin/pulsar-admin functions create \
+  --name parallel-fun \
+  --tenant sample \
+  --namespace ns1 \
+  --py func.py \
+  --className func.ParallelFunction \
+  --parallelism 5
+```
+
 ### Logging
 
-### Data types
+Pulsar Functions created using the [Pulsar Functions SDK(#sdk) can produce logs. Those logs are then published on the Pulsar topic that you specify. The function created using this command, for example, would produce all logs on the `persistent://sample/standalone/ns1/my-func-1-log` topic:
```

Comment:
maybe this should be worded as 'logs can be sent to a log topic using this option?

Reply:
Okay, that makes sense